libcrux-ml-kem/tests/self.rs

```diff
@@ -150,17 +152,30 @@ macro_rules! impl_consistency_incremental {
             let mut pk2_bytes = [0u8; pk2_len()];
             key_pair.pk2_bytes(&mut pk2_bytes);
 
+            // Check that the keys are the same
+            assert_eq!(pk1_bytes, &key_pair_bytes[0..64]);
+            assert_eq!(pk2_bytes, &key_pair_bytes[64..64 + pk2_len()]);
+
             // The other party encapsulates to pk1 ...
             let encaps_randomness = random_array();
             let (ct1, ct2, shared_secret) = {
                 let pk1 = PublicKey1::try_from(&pk1_bytes as &[u8]).unwrap();
                 let (ct1, state) = encapsulate1(&pk1, encaps_randomness);
 
+                // encaps1 with serialized state
+                let mut serialized_state = vec![0u8; encaps_state_len()];
+                let ct12 = encapsulate1_serialized(&pk1, encaps_randomness, &mut serialized_state);
+                assert_eq!(ct1.value, ct12.value);
+
                 // ... and then to pk2.
                 // pk2 is passed in as bytes because the deserializaiton is runtime
                 // platform dependent.
                 let ct2 = encapsulate2(state.as_ref(), &pk2_bytes).unwrap();
 
+                // encaps2 with serialized state
+                let ct22 = encapsulate2_serialized(&serialized_state, &pk2_bytes).unwrap();
+                assert_eq!(ct2.value, ct22.value);
```

Comment:
how does one extract the shared secret from serialized state?

Reply:
That's further down, the `state.shared_secret()`.